flexmeasures/ui/templates/crud/assets.html

```diff
@@ -6,6 +6,83 @@
 
 {% block divs %}
 
+<script>
+
+const asset_icon_map = JSON.parse('{{ asset_icon_map | tojson | safe }}');
+
+function Asset ( id, name, asset_type, sensors, account, latitude, longitude) {
+    let icon = asset_icon_map[asset_type.toLowerCase()];
+    if (icon === undefined)
+      icon = `icon-${asset_type}`
+
+    this.name = `
+      <i class="${ icon } left-icon">${ name }</i>
+    `;
+
+    this.id = id;
+    this.location = "";
+    this.url = `/assets/${id}`
+    this.status = `
+      <a href="/assets/${id}/status">
+        <button type="button" class="btn btn-primary">Status</button>
+      </a>
+    `;
+
+    if (account == null) this.owner = "PUBLIC";
+    else
+      this.owner = `
+        <a href="/accounts/${ account['id'] }" title="View this account">${ account["name"] }</a>
+      `
+
+    this.num_sensors = sensors.length;
+
+    if (latitude !=  null && longitude != null) this.location = `LAT: ${latitude}, LONG: ${longitude}`;
+};
+
+$(document).ready(function() {  
+
+  /*
+  Initialize a DataTable on an HTML element with the ID #assetTable, enabling server-side processing. 
+  The table is configured to display columns for the asset's name, ID, account owner, location, number of sensors, status, and a hidden URL field. 
+  The data for the table is fetched from a server using AJAX.
+  */
+  $("#assetTable").dataTable({
+    serverSide: true,
```

Comment:
I found that when you type your search filter reasonably fast, e.g. `epex`, only the first `e` is used in the filter  in the call to the server. I experimented a bit with adding a line with `searchDelay: 1000,`, which solves the issue for fast typing, but introduces the issue for slow typing.

Here are my other options:
- Try [debounce](https://datatables.net/reference/api/DataTable.util.debounce())
- Try [throttle](https://datatables.net/reference/api/DataTable.util.throttle())
- Try [delayed search](https://stackoverflow.com/questions/5548893/jquery-datatables-delay-search-until-3-characters-been-typed-or-a-button-clicke)
- Accept the bug and move on with our lives, which I consider the best choice for the scope of this PR.

Reply:
I would opt for throttle if this becomes a problem, good suggestion. However, I could not really reproduce this being a problem. I always got a request for the first letter, and then also one for the last of my quick type. See screenshot. I suggest we don't pursue this within this PR.

![Screenshot from 2024-09-08 19-09-52](https://github.com/user-attachments/assets/63f1ec94-1f99-4550-8f84-e0aa8378f59f)